core/src/main/java/com/linecorp/armeria/server/file/HttpFileService.java

```diff
@@ -98,16 +100,18 @@ public static HttpFileService forVfs(HttpVfs vfs) {
 
     private final HttpFileServiceConfig config;
 
-    /**
-     * An LRU cache map that releases the buffer that contains the cached content.
-     */
-    private final Map<String, CachedEntry> cache;
+    private final LoadingCache<PathAndEncoding, CachedEntry> cache;
 
     HttpFileService(HttpFileServiceConfig config) {
         this.config = requireNonNull(config, "config");
 
         if (config.maxCacheEntries() != 0) {
-            cache = Collections.synchronizedMap(new LruMap<String, CachedEntry>(config.maxCacheEntries()));
+            cache = Caffeine.newBuilder()
+                            .maximumSize(config.maxCacheEntries())
+                            .recordStats()
+                            .build(this::getEntryWithoutCache);
+            CacheMetricsCollector collector = new CacheMetricsCollector().register();
+            collector.addCache(getClass().getSimpleName(), cache);
```

Comment:
That's a good question. We could use the `PathMapping.metricName()` + '.vfsCache' or something?